solidity/test/GovernanceTest.js

```diff
@@ -180,50 +196,99 @@ describe("TBTCSystem governance", async function() {
         "0x0000000000000000000000000000000000000002",
+        "0x0000000000000000000000000000000000000003",
       )
 
+      const finalizationTime = await tbtcSystem.getRemainingKeepFactoriesUpdateTime()
+      await increaseTime(finalizationTime.subn(1))
+
       // Should not revert.
-      await tbtcSystem.beginKeepFactorySingleShotUpdate(
+      await tbtcSystem.beginKeepFactoriesUpdate(
+        "0x0000000000000000000000000000000000000005",
+        "0x0000000000000000000000000000000000000006",
+        "0x0000000000000000000000000000000000000007",
+      )
+
+      expect(await tbtcSystem.getRemainingKeepFactoriesUpdateTime()).to.eq.BN(
+        await tbtcSystem.getGovernanceTimeDelay(),
+      )
+    })
+
+    it("does not revert if finalizeKeepFactoriesUpdate has already been called", async () => {
+      await tbtcSystem.beginKeepFactoriesUpdate(
         "0x0000000000000000000000000000000000000001",
         "0x0000000000000000000000000000000000000002",
+        "0x0000000000000000000000000000000000000003",
+      )
+
+      const finalizationTime = await tbtcSystem.getRemainingKeepFactoriesUpdateTime()
+      await increaseTime(finalizationTime.addn(1))
+
+      await tbtcSystem.finalizeKeepFactoriesUpdate()
+
+      // Should not revert.
+      await tbtcSystem.beginKeepFactoriesUpdate(
+        "0x0000000000000000000000000000000000000001",
+        "0x0000000000000000000000000000000000000002",
+        "0x0000000000000000000000000000000000000003",
       )
     })
 
-    it("reverts if finalizeKeepFactorySingleShotUpdate has already been called", async () => {
-      await tbtcSystem.beginKeepFactorySingleShotUpdate(
+    it("does not revert finalizeKeepFactoriesUpdate if upgradeability period has passed", async () => {
+      await tbtcSystem.beginKeepFactoriesUpdate(
         "0x0000000000000000000000000000000000000001",
         "0x0000000000000000000000000000000000000002",
+        "0x0000000000000000000000000000000000000003",
       )
 
-      const finalizationTime = await tbtcSystem.getRemainingKeepFactorySingleShotUpdateTime()
-      await increaseTime(finalizationTime.toNumber() + 1) // 10 days
-      await tbtcSystem.finalizeKeepFactorySingleShotUpdate()
+      const upgradeabilityTime = await tbtcSystem.getRemainingKeepFactoriesUpgradeabilityTime()
+      await increaseTime(upgradeabilityTime.addn(1))
+
+      await tbtcSystem.finalizeKeepFactoriesUpdate()
+    })
+
+    it("reverts for beginKeepFactoriesUpdate if upgradeability period has passed", async () => {
+      await tbtcSystem.beginKeepFactoriesUpdate(
+        "0x0000000000000000000000000000000000000001",
+        "0x0000000000000000000000000000000000000002",
+        "0x0000000000000000000000000000000000000003",
+      )
+
+      const finalizationTime = await tbtcSystem.getRemainingKeepFactoriesUpdateTime()
+      await increaseTime(finalizationTime.addn(1))
+
+      await tbtcSystem.finalizeKeepFactoriesUpdate()
```

Comment:
Why do we need those lines in this test? Isn't it enough to do what's below?

Reply:
I believe it was removed as in https://github.com/keep-network/tbtc/pull/724#discussion_r477294570

Reply:
Oh, nice!